kbase-extension/static/kbase/js/widgets/appWidgets2/input/subdataMethods/manager.js

```diff
@@ -71,158 +87,150 @@ define([
+                                    case 'string':
+                                    case 'number':
+                                        if (selectionId === 'value') {
+                                            id = datum;
+                                        } else {
                                             id = key;
-                                    }
-                                } else {
-                                    id = key;
+                                        }
+                                        break;
+                                    default:
+                                        id = key;
                                 }
+                            } else {
+                                id = key;
+                            }
 
-                                values.push({
-                                    id: id,
-                                    desc: descriptionTemplate(datum), // todo
-                                    // desc: id,
-                                    objectRef: [result.info[6], result.info[0], result.info[4]].join('/'),
-                                    objectName: result.info[1]
-                                });
+                            values.push({
+                                id: id,
+                                desc: descriptionTemplate(datum), // todo
+                                // desc: id,
+                                objectRef: [result.info[6], result.info[0], result.info[4]].join('/'),
+                                objectName: result.info[1]
                             });
-                        }
-                    });
-                    return values.map(function(item) {
-                        item.text = makeLabel(item, arg.spec.ui.showSourceObject);
-                        return item;
-                    });
-                })
-                .then(function(data) {
-                    // sort by id now.
-                    data.sort(function(a, b) {
-                        if (a.id > b.id) {
-                            return 1;
-                        }
-                        if (a.id < b.id) {
-                            return -1;
-                        }
-                        return 0;
-                    });
-                    return data;
+                        });
+                    }
                 });
+                return values.map(function (item) {
+                    item.text = makeLabel(item, arg.spec.ui.showSourceObject);
+                    return item;
+                });
+            };
+            if (subdataSelection.service_function) {
+                dataCall = genericClientCall(subdataSelection, subObjectIdentity);
+            } else {
+                dataCall = workspaceCall(subObjectIdentity);
+            }
+            let subdata_path = subdataSelection.path_to_subdata[0];
+            if (subdata_path.startsWith("WSREF(")) {
```

Comment:
not a big deal, but better to set WSREF as a constant, so that the slice below is not vulnerable to changes in it. Not likely in this deep code, but e.g. someone might want to change it to UPA or whatever and not know that they need to change the 6 below to a 4.